### ISSUE TITLE
When using @ExtendWith(ReportPortalExtension.class) on multiple test classes the test reporting fail with error 500

### ISSUE THREAD

**Issue by b1zzu** [reportportal/agent-java-junit5#83]
reportportal (API): 5.3.3
agent-java-junit5: 5.04

**What**
I created two test classes:
```java
package io.managed.services.test;

import com.epam.reportportal.junit5.ReportPortalExtension;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import static org.junit.jupiter.api.Assertions.assertTrue;

@Tag("fake")
@ExtendWith(ReportPortalExtension.class)
public class FakeOneTest {
    private static final Logger LOGGER = LogManager.getLogger(FakeOneTest.class);

    @Test
    void testOne() {
        LOGGER.info("test one hello");
    }

    @Test
    void testOneFail() throws InterruptedException {
        for (var i = 0; i < 100; i++) {
            Thread.sleep(10);
        }
        LOGGER.info("i is something like: {}", "bla");
        assertTrue(false, "just fail");
    }
}
```
```java
package io.managed.services.test;

import com.epam.reportportal.junit5.ReportPortalExtension;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@Tag("fake")
@ExtendWith(ReportPortalExtension.class)
public class FakeTwoTest {
    private static final Logger LOGGER = LogManager.getLogger(FakeOneTest.class);

    @Test
    void testTwo() {
        LOGGER.info("test one hello");
    }
}
```
and when I execute (both from mvn and intellij) them this is the log output:
```
[INFO] --- maven-failsafe-plugin:3.0.0-M5:integration-test (default) @ mk-e2e-test-suite ---
[WARNING] useSystemClassLoader setting has no effect when not forking
[WARNING] The parameter forkCount should likely not be 0, not forking a JVM for tests reduce test accuracy, ensure to have a <forkCount> >= 1.
Mar 10, 2021 8:30:21 PM org.junit.platform.launcher.core.LauncherConfigurationParameters loadClasspathResource
INFO: Loading JUnit Platform configuration parameters from classpath resource [file:/home/dbizzarr/Projects/github.com/bf2fc6cc711aee1a0c2a/e2e-test-suite/target/test-classes/junit-platform.properties].
[INFO] Running io.managed.services.test.FakeTwoTest
2021-03-10 19:30:22 INFO  [1] [Launch:122] Rerun: false
log4j:WARN No appenders could be found for logger (rp.org.apache.http.client.protocol.RequestAddCookies).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
2021-03-10 19:30:22 INFO  [1] [FakeOneTest:18] test one hello
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.521 s - in io.managed.services.test.FakeTwoTest
[INFO] Running io.managed.services.test.FakeOneTest
2021-03-10 19:30:22 INFO  [1] [Launch:122] Rerun: false
2021-03-10 19:30:23 ERROR [40] [Launch:43] [40] ReportPortal execution error
com.epam.reportportal.exception.ReportPortalException: Report Portal returned error
Status code: 500
Status message: Internal Server Error
Error Message: Unclassified error [org.hibernate.exception.ConstraintViolationException: could not execute statement]
Error Type: UNCLASSIFIED_ERROR

        at com.epam.reportportal.service.ReportPortalErrorHandler.handleError(ReportPortalErrorHandler.java:85) ~[client-java-5.0.18.jar:?]
        at com.epam.reportportal.service.ReportPortalErrorHandler.handle(ReportPortalErrorHandler.java:61) ~[client-java-5.0.18.jar:?]
        at com.epam.reportportal.restendpoint.http.HttpClientRestEndpoint$1.subscribe(HttpClientRestEndpoint.java:526) [client-java-5.0.18.jar:?]
        at io.reactivex.internal.operators.maybe.MaybeCreate.subscribeActual(MaybeCreate.java:45) [rxjava-2.2.10.jar:?]
        at io.reactivex.Maybe.subscribe(Maybe.java:4290) [rxjava-2.2.10.jar:?]
        at io.reactivex.internal.operators.maybe.MaybeSubscribeOn$SubscribeTask.run(MaybeSubscribeOn.java:54) [rxjava-2.2.10.jar:?]
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38) [rxjava-2.2.10.jar:?]
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26) [rxjava-2.2.10.jar:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
2021-03-10 19:30:23 ERROR [40] [Launch:43] [40] ReportPortal execution error
com.epam.reportportal.exception.ReportPortalException: Report Portal returned error
Status code: 500
Status message: Internal Server Error
Error Message: Unclassified error [org.hibernate.exception.ConstraintViolationException: could not execute statement]
Error Type: UNCLASSIFIED_ERROR

        at com.epam.reportportal.service.ReportPortalErrorHandler.handleError(ReportPortalErrorHandler.java:85) ~[client-java-5.0.18.jar:?]
        at com.epam.reportportal.service.ReportPortalErrorHandler.handle(ReportPortalErrorHandler.java:61) ~[client-java-5.0.18.jar:?]
        at com.epam.reportportal.restendpoint.http.HttpClientRestEndpoint$1.subscribe(HttpClientRestEndpoint.java:526) [client-java-5.0.18.jar:?]
        at io.reactivex.internal.operators.maybe.MaybeCreate.subscribeActual(MaybeCreate.java:45) [rxjava-2.2.10.jar:?]
        at io.reactivex.Maybe.subscribe(Maybe.java:4290) [rxjava-2.2.10.jar:?]
        at io.reactivex.internal.operators.maybe.MaybeSubscribeOn$SubscribeTask.run(MaybeSubscribeOn.java:54) [rxjava-2.2.10.jar:?]
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38) [rxjava-2.2.10.jar:?]
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26) [rxjava-2.2.10.jar:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
2021-03-10 19:30:23 INFO  [1] [FakeOneTest:29] i is something like: bla
2021-03-10 19:30:23 INFO  [1] [FakeOneTest:21] test one hello
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1.067 s <<< FAILURE! - in io.managed.services.test.FakeOneTest
[ERROR] io.managed.services.test.FakeOneTest.testOneFail  Time elapsed: 1.034 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: just fail ==> expected: <true> but was: <false>
        at io.managed.services.test.FakeOneTest.testOneFail(FakeOneTest.java:30)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   FakeOneTest.testOneFail:30 just fail ==> expected: <true> but was: <false>
[INFO] 
[ERROR] Tests run: 3, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-failsafe-plugin:3.0.0-M5:verify (default) @ mk-e2e-test-suite ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.683 s
[INFO] Finished at: 2021-03-10T20:30:24+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:3.0.0-M5:verify (default) on project mk-e2e-test-suite: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/dbizzarr/Projects/github.com/bf2fc6cc711aee1a0c2a/e2e-test-suite/target/failsafe-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

```
as you can see when reporting the second Test class it fails with this error:
```
com.epam.reportportal.exception.ReportPortalException: Report Portal returned error
Status code: 500
Status message: Internal Server Error
Error Message: Unclassified error [org.hibernate.exception.ConstraintViolationException: could not execute statement]
Error Type: UNCLASSIFIED_ERROR
```
I tried to run the debugger and I found out that it fails because is trying to create a new Launch with the same UUID (because it read the UUID from .lock or .sync)

**Workaround**

The only workaround that I found for now is setting `junit.jupiter.extensions.autodetection.enabled=true` to enable the ReportPortalExtension globally

**Comment by HardNorth** [comment]
@b1zzu Thanks for reporting this! Yes, the whole mechanism with ` .lock` and `.sync` files doesn't work well. I'm already in the process of re-thinking this.

Another thing is that I'm actually going to remove `@ExtendWith` mention from documentation and leave only service-location configuration as the only recommended way:
https://github.com/reportportal/agent-java-junit5#register-reportportal-extension-through-service-location

**Comment by HardNorth** [comment]
@b1zzu And unfortunately the fix for this is going to be in the final 5.1 agent version, which is still no ETA.

**Comment by b1zzu** [comment]
Thanks @HardNorth 

> @b1zzu And unfortunately the fix for this is going to be in the final 5.1 agent version, which is still no ETA.
We will keep using the autodetect and fortunately, we don't have a case where we don't want to report some tests

**Comment by HardNorth** [comment]
@b1zzu I reverted the change which caused your issue, please use: https://github.com/reportportal/agent-java-junit5/releases/tag/5.0.5
